### PR TITLE
feat 1358: Add statuses to roles

### PR DESCRIPTION
**Pull request by stephanieboyd**

### CODE REVIEW

File: core/builtins/custom_tags.py [hunk 1]
```diff
@@ -248,6 +248,12 @@ def sentence_case(value):
     return " ".join([words[0].capitalize()] + words[1:])
 
 
+def lowercase_and_underscore(value):
+    """Change value to uppercase on initial word and preserve casing on all other words. """
```
Comment (Mark-Spiers-kainos) [inline core/builtins/custom_tags.py]: the string and function name kinda clash

Comment (stephanieboyd) [reply core/builtins/custom_tags.py]: copy paste error, fixed

File: core/services.py [hunk 2]
```diff
@@ -58,10 +61,40 @@ def get_countries(request, convert_to_options=False):
...
 
 
+def get_statuses_as_options(request):
+    """ Get a list of statuses as Options. """
+    data, _ = get_statuses(request)
+    return [Option(key=item["id"], value=sentence_case(item["status"])) for item in data.get("statuses")]
+
+
+def get_permissible_statuses(request, case_id):
+    """ Get a list of case statuses permissible for the user's role. """
```
Comment (Mark-Spiers-kainos) [inline core/services.py]: would it be possible to pass the case_type in here rather than case_id?

File: cases/services.py [hunk 3]
```diff
@@ -42,6 +42,7 @@ def put_case(request, pk, json):
 
 # Applications
 def put_application_status(request, pk, json):
+    json = format_status_in_request_data(json)
```
Comment (andytaylorkainos) [inline cases/services.py]: howcome we have to copy the data in `format_status_in_request_data` if we are overwriting the original data on this line anyway?

Comment (stephanieboyd) [reply cases/services.py]: Cause the original object is immutable, we're just reassigning it back into the name `json`.

File: cases/views/main.py [hunk 4]
```diff
@@ -177,37 +170,22 @@ class ManageCase(TemplateView):
...
         case_id = str(kwargs["pk"])
         case = get_case(request, case_id)
-        statuses, _ = get_statuses(request)
-
-        reduced_statuses = {
-            "statuses": [
-                status for status in statuses["statuses"] if status["status"] in ["closed", "submitted", "withdrawn"]
-            ]
-        }
+        statuses = get_permissible_statuses(request, case_id)
```
[the remark this answers is not in the document]
Comment (stephanieboyd) [reply cases/views/main.py]: Yeah, good suggestion. Changed :)

File: core/services.py (hunk 2 above)
Comment (stephanieboyd) [reply core/services.py]: Good suggestion, thanks

**Pull request merged**